compiler/p/env/OMRCPU.cpp

```diff
@@ -45,23 +152,109 @@ OMR::Power::CPU::isTargetWithinIFormBranchRange(intptr_t targetAddress, intptr_t
           range >= self()->maxIFormBranchBackwardOffset();
    }
 
-bool 
-OMR::Power::CPU::getSupportsHardwareSQRT()
+bool
+OMR::Power::CPU::supportsFeature(uint32_t feature)
    {
-   TR_ASSERT(self()->id() >= TR_FirstPPCProcessor && self()->id() <= TR_LastPPCProcessor, "Not a valid PPC Processor Type");
-   return self()->id() >= TR_FirstPPCHwSqrtProcessor;
+   if (TR::Compiler->omrPortLib == NULL)
+      {
+      return false;
+      }
+
+   OMRPORT_ACCESS_FROM_OMRPORT(TR::Compiler->omrPortLib);
+   return (TRUE == omrsysinfo_processor_has_feature(&_processorDescription, feature));
    }
 
 bool
-OMR::Power::CPU::getSupportsHardwareRound()
+OMR::Power::CPU::is(OMRProcessorArchitecture p)
    {
-   TR_ASSERT(self()->id() >= TR_FirstPPCProcessor && self()->id() <= TR_LastPPCProcessor, "Not a valid PPC Processor Type");
-   return self()->id() >= TR_FirstPPCHwRoundProcessor;
+   if (TR::Compiler->omrPortLib == NULL)
+      return self()->id() == self()->get_old_processor_type_from_new_processor_type(p);
+
+   TR_ASSERT_FATAL((_processorDescription.processor == p) == (self()->id() == self()->get_old_processor_type_from_new_processor_type(p)), "is test %d failed, id() %d, _processorDescription.processor %d", p, self()->id(), _processorDescription.processor);
```

Comment:
do you intend to get rid of this confusing stuffs, processor, old_processor, new_processor eventually?

Reply:
Sorry for the confusion. The old versions will eventually be removed. I wish we can just remove them now.
The issue we have now is omr doesn't have access to the omr port lib so OMR by itself is stuck with the old processor detection implementations. OpenJ9 does have omr port lib so it can start using the new api's. Eventually omr will have access to the port lib as well so the old versions can be removed.

Reply:
The purpose of this PR is to unify the processor detection system across all platforms.